include/unifex/create.hpp

```diff
@@ -1,197 +1,303 @@
- * 
- * \em Example:
- * \code
- *  // A void-returning C-style async API that accepts a context and a continuation:
- *  using callback_t = void(void* context, int result);
- *  void old_c_style_api(int a, int b, void* context, callback_t* callback_fn);
- *
- *  // A sender-based async API implemented in terms of the C-style API (using C++20):
- *  unifex::typed_sender auto new_sender_api(int a, int b) {
- *    return unifex::create<int>([=](auto& rec) {
- *      old_c_style_api(a, b, &rec, [](void* context, int result) {
- *        unifex::void_cast<decltype(rec)>(context).set_value(result);
- *      });
- *    });
- *  }
- * \endcode
- * \tparam ValueTypes The value types of the resulting sender. Should be the list of
- *   value type(s) accepted by the callback (with the exception of the void*
- *   context).
- * \param[in] fn A void-returning callable that accepts an lvalue reference to an object
- *   whose type satisfies the \c unifex::receiver_of<ValueTypes...> concept. This function
- *   should dispatch to the C-style callback (see example).
- * \param[in] ctx An optional extra bit of data to be bundled with the receiver passed to
- *   \c fn. E.g., if \c fn is a lambda that accepts <tt>(auto& rec)</tt> and \c ctx is 42,
- *   then from within the body of \c fn , the value of the expression \c rec.context() is
- *   42.
- * \return A sender that, when connected and started, dispatches to the wrapped C-style
- *   API with the callback of your choosing. The receiver passed to \c fn wraps the
- *   receiver passed to \c connect . Your callback should "complete" the receiver passed
- *   to \c fn , which will complete the receiver passed to \c connect in turn.
- */
-template <typename... ValueTypes>
-inline constexpr _create::_cpo::_fn<ValueTypes...> create {};
-
-using _create::void_cast;
-} // namespace unifex
-
-#include <unifex/detail/epilogue.hpp>
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/manual_lifetime.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/type_traits.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _create {
+
+template <typename Receiver, typename Fn, typename... Context>
+struct _op {
+  struct type;
+  struct _receiver {
+    explicit _receiver(type* op) noexcept(
+        std::is_nothrow_move_constructible_v<Receiver>)

```

Comment:
This looks unconditionally `noexcept`; copying a pointer into a member can't throw, and I don't see any other operations here.

```suggestion
    explicit _receiver(type* op) noexcept
```